luajit-2.1/src/luaconf.h

```diff
@@ -135,7 +138,7 @@
 #define LUA_API		__declspec(dllimport)
 #endif
 #else
-#define LUA_API		extern
+#define LUA_API		extern "C"
```

Comment:
I dont think we should be messing around wit hthe original lua source code?  We can simply add extern "C" in our own c++ header files?
```
#ifdef __cplusplus
extern "C"
#endif
    #include "luaconf.h"
#ifdef __cplusplus
}
#endif
```